fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneMapQueryTest.java

```diff
@@ -0,0 +1,461 @@
+    @Override
+    protected RecordLayerPropertyStorage.Builder addDefaultProps(final RecordLayerPropertyStorage.Builder props) {
+        return super.addDefaultProps(props)
+                .addProp(LuceneRecordContextProperties.LUCENE_EXECUTOR_SERVICE, (Supplier<ExecutorService>)() -> executorService);
+    }
+
+    protected void openRecordStore(FDBRecordContext context) {
+        openRecordStore(context, md -> {
+        }, SIMPLE_TEXT_SUFFIXES);
+    }
+
+    protected void openRecordStore(FDBRecordContext context, RecordMetaDataHook hook, Index simpleDocIndex) {
+        RecordMetaDataBuilder metaDataBuilder = RecordMetaData.newBuilder().setRecords(TestRecordsTextProto.getDescriptor());
+        metaDataBuilder.getRecordType(TextIndexTestUtils.COMPLEX_DOC).setPrimaryKey(concatenateFields("group", "doc_id"));
+        if (simpleDocIndex != null) {
+            metaDataBuilder.removeIndex("SimpleDocument$text");
+            metaDataBuilder.addIndex(TextIndexTestUtils.SIMPLE_DOC, simpleDocIndex);
+        }
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_2_LONG_LUCENE_INDEX);
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_WRAPPER_TO_LONG_LUCENE_INDEX);
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_2_LONG_WRAPPER_LUCENE_INDEX);
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_2_INT_LUCENE_INDEX);
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_2_DOUBLE_LUCENE_INDEX);
+        hook.apply(metaDataBuilder);
+        recordStore = getStoreBuilder(context, metaDataBuilder.getRecordMetaData())
+                .setSerializer(TextIndexTestUtils.COMPRESSING_SERIALIZER)
+                .createOrOpen();
+        setupPlanner(null);
+    }
+
+    private void initializeNested() {
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context);
+            mapDocuments.forEach(recordStore::saveRecord);
+            commit(context);
+        }
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "d, true",
+            "Blah, false",
+            "c, false"
+    })
+    void mapStringToLong(String value, boolean found) throws Exception {
+        initializeNested();
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context);
+
+            RecordQuery query = RecordQuery.newBuilder()
+                    .setRecordType(MAP_DOC)
+                    .setFilter(Query.field("stringToLongMap").matches(Query.field("values").oneOfThem().matches(Query.field("key").equalsValue(value))))
+                    .build();
+            RecordQueryPlan plan = planQuery(query);
+            try (RecordCursor<FDBQueriedRecord<Message>> recordCursor = recordStore.executeQuery(plan)) {
+                List<Long> primaryKeys = recordCursor.map(FDBQueriedRecord::getPrimaryKey).map(t -> t.getLong(0)).asList().get();
+                final Set<Long> expected = found ? Set.of(0L, 1L, 2L) : Set.of();
+                assertEquals(expected, Set.copyOf(primaryKeys));
```

Comment:
Should these assert that the correct index is used?

Reply:
Done.